tests/schemables/WithDate.dateFromString.test.ts

```diff
@@ -98,47 +103,50 @@ describe('dateFromIsoString', () => {
   describe('Guard', () => {
     test.each(valid)('validates valid date strings, %s', str => {
       const test = new Date(str)
-      const result = ISODateString.Guard.is(test)
+      const result = ISODateString.Guard.dateFromString.is(test)
       expect(result).toBe(true)
     })
     test.each(invalid)('invalidates invalid date strings, %s', str => {
       const test = new Date(str)
-      const result = ISODateString.Guard.is(test)
+      const result = ISODateString.Guard.dateFromString.is(test)
       expect(result).toBe(false)
     })
   })
 
   describe('TaskDecoder', () => {
     test.each(valid)('validates valid date strings, %s', async str => {
-      const result = await ISODateString.TaskDecoder.decode(str)()
+      const result = await ISODateString.TaskDecoder.dateFromString.decode(str)()
       expect(result._tag).toBe('Right')
     })
     test.each(invalid)('invalidates invalid date strings, %s', async str => {
-      const result = await ISODateString.TaskDecoder.decode(str)()
+      const result = await ISODateString.TaskDecoder.dateFromString.decode(str)()
       expect(result._tag).toBe('Left')
     })
   })
 
   describe('Type', () => {
     test.each(valid)('validates valid date strings, %s', str => {
-      const result = ISODateString.Type.decode(str)
+      const result = ISODateString.Type.dateFromString.decode(str)
       expect(result._tag).toBe('Right')
     })
     test.each(invalid)('invalidates invalid date strings, %s', str => {
-      const result = ISODateString.Type.decode(str)
+      const result = ISODateString.Type.dateFromString.decode(str)
       expect(result._tag).toBe('Left')
     })
   })
 
   describe('Arbitrary', () => {
     it('generates valid ISODateString', () => {
-      validateArbitrary(ISODateString, ISODateString.Guard.is)
+      validateArbitrary(
+        { Arbitrary: ISODateString.Arbitrary.dateFromString },
+        ISODateString.Guard.dateFromString.is,
+      )
```

Comment:
I think, where we have an option to, we should write this so it doesn't use the generated `Guard` instance. especially with both `Arbitrary` and `Guard` both being derived from a `Pattern`, this is essentially more a test of `Pattern` than it is of the actual arbitrary.

in some cases I'm not sure of a better option. here, though, we can use `Date` to make sure what it produces is valid